lib/resty/mlcache.lua

```diff
@@ -513,7 +545,12 @@ function _M:get(key, opts, cb, ...)
 
     -- set our own worker's LRU cache
 
-    set_lru(self, key, data, ttl, neg_ttl)
+    data, err = set_lru(self, key, data, ttl, neg_ttl, l1_serializer)
+    if data == CACHE_MISS_SENTINEL_LRU then
+        data = nil -- convert misses back to nil for return values
+    elseif err then
+        return unlock_and_ret(lock, nil, err)
+    end
```

Comment:
We might save some cycles (and stay consistent) if we take a similar approach as the above L2 hit here:
```lua
data, err = set_lru(self, key, data, ttl, neg_ttl, l1_serializer)
if err then
    return unlock_and_ret(lock, nil, err)
end

if data == CACHE_MISS_SENTINEL_LRU then
    return unlock_and_ret(lock, nil, nil, 2)
end

return unlock_and_ret(lock, data, nil, 2)
```

This way we avoid assigning `data = nil` but keep the same amount of branches in the execution path?